### ISSUE TITLE
Feature request: graceful malformed packet drop

### ISSUE THREAD

**Issue by jsmif** [nccgroup/Sniffle#75]
I just went to do a headless long-term sniffing run, and when I got back I saw Sniffle had died prematurely with the following error:

```
Traceback (most recent call last):
  File "/home/user/Sniffle/python_cli/sniff_receiver.py", line 216, in <module>
    main()
  File "/home/user/Sniffle/python_cli/sniff_receiver.py", line 157, in main
    print_message(msg, args.quiet)
  File "/home/user/Sniffle/python_cli/sniff_receiver.py", line 161, in print_message
    print_packet(msg, quiet)
  File "/home/user/Sniffle/python_cli/sniff_receiver.py", line 168, in print_packet
    dpkt = DPacketMessage.decode(pkt)
  File "/home/user/Sniffle/python_cli/packet_decoder.py", line 61, in decode
    return AdvertMessage.decode(pkt)
  File "/home/user/Sniffle/python_cli/packet_decoder.py", line 111, in decode
    return tc(pkt)
  File "/home/user/Sniffle/python_cli/packet_decoder.py", line 275, in __init__
    self.aa_conn = struct.unpack('<L', self.body[14:18])[0]
struct.error: unpack requires a buffer of 4 bytes
```
(Note: the device was accidentally out of date, so those line numbers correspond to commit f4cc588fdc5cad5bd53df55529f581cd6ef647d8).

So the ask would be to wrap instances of struct.unpack(), or anything else which might throw an error on malformed packets in a try-except, to just make it throw away the packet so it doesn't accidentally error out early on long runs. Thank you.

**Comment by sultanqasim** [comment]
Good idea, for long runs there are many situations in which there could be a glitch in the data coming in, for example I’ve seen intermittent cases of chunks of data being lost over USB CDC-ACM, especially during context switches when running inside a VM and under situations of high CPU load.

**Comment by sultanqasim** [comment]
Hmm, did 5cecf125c634962cc72442fb0165513884e3f66e cause that for you?  Does the error appear right at startup? The only way that can happen is if `pkt` [here](https://github.com/nccgroup/Sniffle/blob/5cecf125c634962cc72442fb0165513884e3f66e/python_cli/sniffle_hw.py#L289) is all whitespace or CRLF characters.

If you change (this)[https://github.com/nccgroup/Sniffle/blob/5cecf125c634962cc72442fb0165513884e3f66e/python_cli/sniffle_hw.py#L292] line to `if len(pkt.strip()) < 4:` does that fix it?

**Comment by sultanqasim** [comment]
Try the latest master branch, hopefully that fixes this particular crash you reported.

**Comment by sultanqasim** [comment]
And this should improve general robustness: 87ac1bd0a68a199839675eb4ac0c9b8371a006aa

**Comment by XenoKovah** [comment]
FWIW I left my sniffers (confirmed to be on 1.9.3) running overnight in a hotel last night (so I won't be able to replicate this in the future), and when I got up one of them had got basically the same error as the OP:

```
Traceback (most recent call last):
  File "/home/test/Downloads/Sniffle/python_cli/sniff_receiver.py", line 203, in <module>
    main()
  File "/home/test/Downloads/Sniffle/python_cli/sniff_receiver.py", line 162, in main
    print_message(msg, args.quiet)
  File "/home/test/Downloads/Sniffle/python_cli/sniff_receiver.py", line 166, in print_message
    print_packet(msg, quiet)
  File "/home/test/Downloads/Sniffle/python_cli/sniff_receiver.py", line 173, in print_packet
    dpkt = DPacketMessage.decode(pkt, hw.decoder_state)
  File "/home/test/Downloads/Sniffle/python_cli/packet_decoder.py", line 81, in decode
    dpkt = AdvertMessage.decode(pkt, dstate)
  File "/home/test/Downloads/Sniffle/python_cli/packet_decoder.py", line 146, in decode
    return tc(pkt)
  File "/home/test/Downloads/Sniffle/python_cli/packet_decoder.py", line 310, in __init__
    self.aa_conn = struct.unpack('<L', self.body[14:18])[0]
struct.error: unpack requires a buffer of 4 bytes
```

**Comment by sultanqasim** [comment]
That should be fixed in the latest master branch; it will still work fine with the v1.9.3/v1.9.2 firmware.

**Comment by XenoKovah** [comment]
Ah, I didn't pay attention and realize the cited fix commit wasn't in 1.9.3. Thanks
